cedar-policy/src/api.rs

```diff
@@ -1258,22 +1258,21 @@ impl Validator {
     }
 
     #[cfg(feature = "level-validate")]
-    /// Validate all policies in a policy set, collecting all validation errors
-    /// found into the returned `ValidationResult`. Include level validation (RFC 76).
-    /// Each error is returned together with the policy id of the policy where the error
-    /// was found. If a policy id included in the input policy set does not appear in the
-    /// output iterator, then that policy passed the validator. If the function
+    /// Strictly validate all policies in a policy set, collecting all validation errors
+    /// found into the returned `ValidationResult`. If strict validation passes, run level
+    /// validation (RFC 76). Each error is returned together with the policy id of the policy
+    /// where the error was found. If a policy id included in the input policy set does not
+    /// appear in the output iterator, then that policy passed the validator. If the function
     /// `validation_passed` returns true, then there were no validation errors found, so
     /// all policies in the policy set have passed the validator.
-    pub fn validate_with_level(
```

Comment:
What was the reason for this change? I don't see how the algorithm would rely on strict typechecking in particular.

I'd rather keep the old API with the `mode` param if that possible. Not a big deal otherwise

Reply:
I don't see why it wouldn't work with permissive validation, but I haven't thought about it enough (and more importantly haven't tested it) so I'd rather merge this and add permissive validation in the future.